src/async-wrap.cc

```diff
@@ -38,6 +38,8 @@ static void SetupHooks(const FunctionCallbackInfo<Value>& args) {
   //   non-zero to have those callbacks called. This only affects the init
   //   callback. If the init callback was called, then the pre/post callbacks
   //   will automatically be called.
+  // - kEnabled (1): Tells the AsyncWrap constructor whether it should
+  //   make any calls to the JS hook functions.
```

Comment:
There are 3 of them right (ìnit`, `pre` and `post`). This should affect all of them.

I will add a test case then. I was just unsure about the test policy for async wrap.